src/merge.ts

```diff
@@ -0,0 +1,135 @@
+import type { Config, Octokit, File, FrontMatter } from './types';
+import type { Repository } from '@octokit/webhooks-types';
+import localConfig from './localConfig';
+import fm from 'front-matter';
+import yaml from 'js-yaml';
+
+async function generateEIPNumber(octokit: Octokit, repository: Repository) {
+    // Get all EIPs
+    const eips = (await octokit.rest.repos.getContent({
+        owner: repository.owner.login,
+        repo: repository.name,
+        path: 'EIPS'
+    })).data as any[];
+
+    // Get all EIP numbers
+    const eipNumbers = eips
+        .filter(eip => eip.name.startsWith('eip-'))
+        .map(eip => Number(eip.name.split('-')[1]));
+
+    // Find the biggest EIP number
+    const eipNumber = Math.max(...eipNumbers);
+
+    // Add a random number from 1-5 to the EIP number
+    // This is to prevent conflicts when multiple PRs are merged at the same time, and to prevent number gaming
+    return eipNumber + Math.floor(Math.random() * 5) + 1;
```

Comment:
I don't believe we need nor should use randomness in this bot. If we choose the number and merge in an atomic operation, there's no way to have a conflict right?

Reply:
The randomness was discussed as a solution to number gaming.

Reply:
Then I guess we should remove the note about conflicts?

---

I'm still not sold on needing randomness. This'll just consume EIP numbers five times faster. Since authors don't control when an editor approves, will they have any way to game the system? Maybe by making `eipw` fail, getting an approval, then fixing the `eipw` errors? Or does editing the PR clear approvals?

Reply:
Editing the PR does clear approvals, but since there's about 1 EIP every 10 numbers, 1 in 5 greatly improves the density. How about 1 in 3?

```suggestion
    // Add a random number from 1-5 to the EIP number
    // This is to prevent conflicts when multiple PRs are merged at the same time, and to prevent number gaming
    return eipNumber + Math.floor(Math.random() * 3) + 1;
```